File: frontend-mealsonwheels/src/types/order/OrderRequestDTO.ts
// src/types/order/OrderRequestDTO.ts

import type { OrderMealDTO } from "./OrderMealDTO";

export interface OrderRequestDTO {
  memberId: number;
  partnerId: number;
  totalAmount: number;
  riderId: number;

  mealIds: number[];
  meals: OrderMealDTO[];
}

File: backend-mealsonwheels/src/main/java/com/merrymeal/mealsonwheels/dto/order/OrderRequestDTO.java
package com.merrymeal.mealsonwheels.dto.order;

import lombok.*;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class OrderRequestDTO {

    private Long caregiverId;
    private Long memberId;
    private Long partnerId;
    private Double totalAmount;
    private Long riderId;

    private List<Long> mealId; // IDs of meals being ordered this is former "mealIds"
    private List<OrderMealDTO> meals; // Detailed info including photos and quantities
}
